File: model_dflowfm_blackbox/tests/lake_kalman/plot_sequential.m
% Sequential simulation:
% Make plot to show difference between observations and predicted values
%
[t, obs, pred_f pred_a]=load_sequential('SequentialSimulation_results');


for j = 1:4,
    gcf=j; figure(gcf);
    plot(pred_f(j,1:length(t)),'DisplayName', 'Simulation');
    hold on
    plot(obs(j,1:length(t)),'k', 'DisplayName', 'Observation');
    title(strcat('S',num2str(j)));
    %legend('prediction','observation');
    xlabel('time');
end


diff=pred_f-obs;
gcf = gcf + 1; figure(gcf);
plot(diff')
title('Difference between model and observations');
ylabel('prediction-observation');
xlabel('time');

[nobs,ntimes]=size(diff)
for iobs=1:nobs
   rms=norm(diff(iobs,:))/sqrt(ntimes);
   disp(['RMS of location ',num2str(iobs),' ',num2str(rms)]);
end

File: model_dflowfm_blackbox/tests/estuary_kalman_FMSuite2019.01_bcfile/plot_sequential.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
"""
   Sequential simulation:
   Make plot to show difference between observations and predicted values
"""

import SequentialSimulation_results as seqsim
import matplotlib.pyplot as plt

obs = seqsim.obs
pred_f = seqsim.pred_f_central
nobs = obs.shape[0]
nstat = obs.shape[1]

for i in range(nstat):
    fig = plt.figure()
    plt.plot(pred_f[:,i])
    plt.plot(obs[:,i])
    plt.title('Station0'+str(i+1))
    plt.xlabel('time')
    plt.legend(['prediction','observation'])

diff = pred_f - obs
fig4 = plt.figure()
plt.plot(diff)
plt.title('Difference between model and observations')
plt.ylabel('prediction-observation')
plt.xlabel('time')
plt.legend(['Station 1', 'Station 2', 'Station 3'])

plt.show()
print("hou plotjes in beeld")

